### PR TITLE
feat(batchRequest): Adds initial batchRequest feature

### PR DESCRIPTION
**Pull request by theodorewahle**

### CODE REVIEW

File: lib/vehicle.js [hunk 1]
```diff
@@ -494,4 +494,21 @@ Vehicle.prototype.unlock = function() {
   return util.wrap(req).then(({status}) => ({status}));
 };
 
+/**
+ * POST Vehicle.batch
+ *
+ * @return {Promise.<Security>} response on successful request
```
Comment (theodorewahle) [inline lib/vehicle.js]:
```suggestion
 * @return {Promise.<Batch>} response on successful request
```

File: lib/vehicle.js [hunk 2]
```diff
@@ -494,4 +484,21 @@ Vehicle.prototype.unlock = function() {
   return util.wrap(req).then(({status}) => ({status}));
 };
 
+/**
+ * POST Vehicle.batch
```
Comment (adolfoportilla) [inline lib/vehicle.js]: Can we add here some examples of the return object and the accepted params?
```
@params {string[]} paths - A list of paths of endpoints to send requests to.
@example 
{
  responses: [....]
}

```

File: test/end-to-end/vehicle.js [hunk 3]
```diff
@@ -58,6 +58,10 @@ test('vehicle unlock', async(t) => {
   await t.notThrowsAsync(t.context.vehicle.unlock());
 });
 
+test('vehicle batch', async(t) => {
+  await t.notThrowsAsync(t.context.vehicle.batch(['/odometer']));
```
Comment (adolfoportilla) [inline test/end-to-end/vehicle.js]:
```suggestion
  await t.notThrowsAsync(t.context.vehicle.batch(['/odometer', '/location']));
```

Comment (adolfoportilla) [reply test/end-to-end/vehicle.js]: Throw in another endpoint to make sure we can hit more than 1 with batch :smile:

File: test/unit/lib/vehicle.js [hunk 4]
```diff
@@ -489,53 +472,112 @@ test('charge - no age', async function(t) {
...
+      },
+      {
+        path: '/fuel',
+      },
+      {
+        path: '/sunroof',
+      },
+    ],
+  };
+  const responseBody = {
```
Comment (adolfoportilla) [inline test/unit/lib/vehicle.js]:
```suggestion
  const expectedResponseBody = {
```

Comment (adolfoportilla) [reply test/unit/lib/vehicle.js]: Rename it so its more clear that this is the expected response body and not the actual response body.

File: test/unit/lib/vehicle.js [hunk 5]
```diff
@@ -489,53 +472,112 @@ test('charge - no age', async function(t) {
...
+      },
+    ],
+  };
+
+  t.context.n = nocks
+    .base()
+    .post('/batch', requestBody)
+    .reply(200, responseBody);
+
+  const response = await vehicle.batch(paths);
```
Comment (adolfoportilla) [inline test/unit/lib/vehicle.js]: nit: Add an extra line after here.

We try to follow the AAA unit testing pattern (https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80), so ideally we separate each section that way its easier to read and review the tests!

Comment (theodorewahle) [reply test/unit/lib/vehicle.js]: Fixed. I definitely agree that we should stick to AAA!

File: lib/vehicle.js (hunk 2 above)
Comment (theodorewahle) [reply lib/vehicle.js]: added

**Pull request merged**